src/components/GlobalPreferences/Notifications/NotificationsManager.js

```diff
@@ -58,46 +59,73 @@ const NotificationsManager = React.memo(function NotificationsManager() {
     }
   }, [account, address, token])
 
+  useEffect(() => {
+    if (token || address) {
+      return
+    }
+    if (!account) {
+      return setStartingScreenId(UNLOCK_NOTIFICATIONS_SCREEN)
+    }
+
+    if (fetchingSubscriptionData || fetchingEmail) {
+      return setStartingScreenId(LOADING_SCREEN)
+    }
+  }, [account, fetchingEmail, fetchingSubscriptionData, token, address])
+
+  useEffect(() => {
+    if (token || address) {
+      return
+    }
+    if (fetchingEmail || fetchingSubscriptionData) {
+      return
+    }
+    if (account && jurorNeedsSignature) {
+      return setStartingScreenId(UNLOCK_NOTIFICATIONS_SCREEN)
+    }
+  }, [account, jurorNeedsSignature, fetchingEmail, fetchingSubscriptionData, token, address])
+
+  useEffect(() => {
+    if (
+      !account ||
+      fetchingSubscriptionData ||
+      fetchingEmail ||
+      jurorNeedsSignature
+    ) {
+      return
+    }
+    if (!emailVerified && emailExists) {
+      return setStartingScreenId(VERIFY_EMAIL_ADDRESS_PREFERENCES)
+    }
+    if (!emailVerified) {
+      return setStartingScreenId(EMAIL_NOTIFICATIONS_FORM_SCREEN)
+    }
+
+    if (emailVerified && !jurorNeedsSignature) {
+      return setStartingScreenId(NOTIFICATIONS_PREFERENCES_SCREEN)
+    }
+  }, [account, emailExists, emailVerified, fetchingEmail, fetchingSubscriptionData, jurorNeedsSignature])
```

Comment:
I'm a little bit confused with the above `useEffect()`s. Should they all live under a single effect? Otherwise it's going to be weird if some conditions trigger multiple of the `useEffect()`s and call `setStartingScreenId()` multiple times.